OpenRA.Mods.Common/Traits/World/Locomotor.cs

```diff
@@ -280,31 +345,143 @@ bool IsBlockedBy(Actor self, Actor otherActor, Actor ignoreActor, CellConditions
 			}
 
 			// If we cannot crush the other actor in our way, we are blocked.
-			if (Crushes.IsEmpty)
+			if (Info.Crushes.IsEmpty)
 				return true;
 
 			// If the other actor in our way cannot be crushed, we are blocked.
 			// PERF: Avoid LINQ.
 			var crushables = otherActor.TraitsImplementing<ICrushable>();
 			foreach (var crushable in crushables)
-				if (crushable.CrushableBy(otherActor, self, Crushes))
+				if (crushable.CrushableBy(otherActor, self, Info.Crushes))
 					return false;
 
 			return true;
 		}
 
-		public virtual bool DisableDomainPassabilityCheck { get { return false; } }
+		static bool IsMovingInMyDirection(Actor self, Actor other)
+		{
+			var otherMobile = other.TraitOrDefault<Mobile>();
```

Comment:
From IRC: replacing this with `other.OccupiesSpace as Mobile` could give a noticeable win.